File: src/main/scala/fi/oph/koski/schema/Ulkomaanjakso.scala
package fi.oph.koski.schema

import java.time.LocalDate

import fi.oph.koski.schema.annotation.{KoodistoUri, Tooltip}
import fi.oph.scalaschema.annotation.Description

@Description("Opintoihin liittyvien ulkomaanjaksojen tiedot. Ulkomaanjakson tiedot sisältävät alku- ja loppupäivämäärät, tiedon siitä, missä maassa jakso on suoritettu, sekä kuvauksen jakson sisällöstä.")
@Tooltip("Opintoihin liittyvien ulkomaanjaksojen tiedot. Ulkomaanjakson tiedot sisältävät alku- ja loppupäivämäärät, tiedon siitä, missä maassa jakso on suoritettu, sekä kuvauksen jakson sisällöstä. Voit lisätä useita ulkomaanjaksoja 'lisää uusi'-painikkeesta.")
case class Ulkomaanjakso(
  alku: LocalDate,
  loppu: Option[LocalDate],
  @Description("Missä maassa jakso on suoritettu")
  @Tooltip("Tieto siitä, missä maassa jakso on suoritettu.")
  @KoodistoUri("maatjavaltiot2")
  maa: Koodistokoodiviite,
  @Description("Kuvaus ulkomaanjakson sisällöstä. Esimerkiksi tieto siitä, opiskeleeko oppija ulkomaisessa oppilaitoksessa vai onko työharjoittelussa tai työssäoppimisessa.")
  @Tooltip("Kuvaus ulkomaanjakson sisällöstä. Esimerkiksi tieto siitä, opiskeleeko oppija ulkomaisessa oppilaitoksessa vai onko työharjoittelussa tai työssäoppimisessa.")
  kuvaus: LocalizedString
) extends Jakso

File: web/app/types/fi/oph/koski/schema/Ulkomaanjakso.ts
import { Koodistokoodiviite } from './Koodistokoodiviite'
import { LocalizedString } from './LocalizedString'

/**
 * Opintoihin liittyvien ulkomaanjaksojen tiedot. Ulkomaanjakson tiedot sisältävät alku- ja loppupäivämäärät, tiedon siitä, missä maassa jakso on suoritettu, sekä kuvauksen jakson sisällöstä.
 *
 * @see `fi.oph.koski.schema.Ulkomaanjakso`
 */
export type Ulkomaanjakso = {
  $class: 'fi.oph.koski.schema.Ulkomaanjakso'
  alku: string
  loppu?: string
  maa: Koodistokoodiviite<'maatjavaltiot2', string>
  kuvaus: LocalizedString
}

export const Ulkomaanjakso = (o: {
  alku: string
  loppu?: string
  maa: Koodistokoodiviite<'maatjavaltiot2', string>
  kuvaus: LocalizedString
}): Ulkomaanjakso => ({ $class: 'fi.oph.koski.schema.Ulkomaanjakso', ...o })

Ulkomaanjakso.className = 'fi.oph.koski.schema.Ulkomaanjakso' as const

export const isUlkomaanjakso = (a: any): a is Ulkomaanjakso =>
  a?.$class === 'fi.oph.koski.schema.Ulkomaanjakso'
